### PR TITLE
Feature/optimization for agilex

### PR DESCRIPTION
**Pull request by wangyon1**

### CODE REVIEW

File: host/src/fpga.cpp [hunk 1]
```diff
@@ -1261,18 +1274,21 @@ void Device::enqueue_input_data_KeySwitch(FPGAObject_KeySwitch* fpga_obj) {
         keyswitch_queues_[KEYSWITCH_LOAD], *(keys->k_switch_keys_1_),
         *(keys->k_switch_keys_2_), *(keys->k_switch_keys_3_),
         fpga_obj->in_objs_.size());
-    const auto& start_ocl = std::chrono::high_resolution_clock::now();
+
     int obj_id = KeySwitch_id_ % 2;
     copyKeySwitchBatch(fpga_obj, obj_id);
+    KeySwitch_events_write_[obj_id][0] = copy_buffer_to_device(
+        keyswitch_queues_[KEYSWITCH_LOAD], *(fpga_obj->mem_t_target_iter_ptr_));
+    KeySwitch_events_write_[obj_id][0].wait();
```
Comment (ymeng-git) [inline host/src/fpga.cpp]: For release, we can comment out line 1280-1281, and add a comment to explain that copy_buffer_to_device() and wait() is a utility to force blocked write, and to facilitate performance measure on FPGA.  The release is to support streaming, and blocking write will slow things down.

**Review by ymeng-git — CHANGES_REQUESTED** [review]
comment out the call to the utility function copy_buffer_to_device().  The rest looks good.

**Comment by ymeng-git** [comment]
Could you enable github signing of your commits with your GPG key?  The unsigned/unverified commits will not go through the merge request.

File: host/src/fpga.cpp (hunk 1 above)
Comment (wangyon1) [reply host/src/fpga.cpp]: Done.

**Review by ymeng-git — APPROVED** [review]

**Review by yzhou79 — APPROVED** [review]

**Pull request merged**